run_node.sh

```diff
@@ -40,35 +40,38 @@ DB_SNAPSHOT_PATH="chains/testnet/"     # testnet by default
 CHAINSPEC_FILE="testnet_chainspec.json"
 
 
-while getopts n:i:-: OPT; do
-    if [ "$OPT" = "-" ]; then   # long option: reformulate OPT and OPTARG
-        OPT="${OPTARG%%=*}"       # extract long option name
-        OPTARG="${OPTARG#$OPT}"   # extract long option argument (may be empty)
-        OPTARG="${OPTARG#=}"      # if long option argument, remove assigning `=`
-    fi
-    echo ""
-    case "$OPT" in
-        help) # display Help
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -h | --help) # display Help
             Help
             exit;;
-        archivist) # Run as an archivist
-            ARCHIVIST=true;;
-        n | name) # Enter a name
-            NAME=$OPTARG;;
-        mainnet) # Join the mainnet
+        --archivist) # Run as an archivist
+            ARCHIVIST=true
+            shift;;
+        -n | --name) # Enter a name
+            NAME="$2"
+            shift 2;;
+        --mainnet) # Join the mainnet
             DB_SNAPSHOT_PATH="chains/mainnet/"
             CHAINSPEC_FILE="mainnet_chainspec.json"
             DB_SNAPSHOT_FILE="db_chain_backup.tar.gz"
-            DB_SNAPSHOT_URL="${MAINNET_DB_SNAPSHOT_URL_BASE}/${DB_SNAPSHOT_FILE}";;
-        i | image) # Enter a base path
-            ALEPH_IMAGE=$OPTARG
-            PULL_IMAGE=false;;
-        build_only)
-            BUILD_ONLY=true;;
-        sync_from_genesis)
-            SYNC=true;;
-        *) # Invalid option
-            echo "Error: Invalid option"
+            DB_SNAPSHOT_URL="${MAINNET_DB_SNAPSHOT_URL_BASE}/${DB_SNAPSHOT_FILE}"
+            shift;;
+        -i | --image) # Enter a base path
+            ALEPH_IMAGE="$2"
+            PULL_IMAGE=false
+            shift 2;;
+        --build_only)
+            BUILD_ONLY=true
+            shift;;
+        --sync_from_genesis)
+            SYNC=true
+            shift;;
+        -* | --* )
```

Comment:
I think lines 70 till 72 are not required since pattern in line 73 would cover that case anyway; also in the future we might want to add support for positional parameters, that are differentiated from other arguments by `--` (e.g. `cargo run --release -- a b c` is an example of positional args `a b c` passed to the binary run by `cargo`)

Reply:
Right now we display a different error when it’s not even an option (I.e. doesn’t start with a dash). We might want to change that in the future, however